shotover/src/hot_reload/server.rs

```diff
@@ -145,20 +142,43 @@ impl UnixSocketServer {
                 );
                 (Response::SendListeningSockets, collected_fds)
             }
-            Request::ShutdownOriginalNode => {
-                info!("Processing ShutdownOriginalNode request - initiating immediate shutdown");
-
-                // Trigger shutdown by sending true through the watch channel
-                if let Err(e) = self.shutdown_tx.send(true) {
-                    error!("Failed to send shutdown signal: {:?}", e);
-                    return (
-                        Response::Error("Failed to trigger shutdown".to_string()),
-                        Vec::new(),
-                    );
+            Request::GradualShutdown => {
+                info!(
+                    "Processing GradualShutdown request - initiating gradual connection draining"
+                );
+
+                // Send gradual shutdown requests to all sources
+                for source in &self.sources {
+                    let (response_tx, response_rx) = tokio::sync::oneshot::channel();
+                    let shutdown_request = GradualShutdownRequest {
+                        return_chan: response_tx,
```

Comment:
Should we remove this return_chan?
I dont think it provides any value, since any source that has received the request will immediately respond, and we can already tell which sources wont receive it by checking for the error returned by `source.gradual_shutdown_tx.send(shutdown_request)` which we already check for.

Reply:
We could change the return_chan logic to instead indicate when the source has completed shutdown (instead of just started shutting down). But I dont think there's any need for this logic to block on the sources shutting down.